services/QuillLMS/lib/tasks/temporary/users.rake

```diff
@@ -0,0 +1,18 @@
+namespace :users do
+  task :update_email_domain, [:old_domain, :new_domain] => [:environment] do |task, args|
+    users = User.where("email ILIKE ?", "%@#{args[:old_domain]}")
+
+    puts "Going to update #{users.count} email addresses"
+    num_updates = 0
+
+    ActiveRecord::Base.transaction do
+      users.find_each do |user|
+        user.update(email: user.email.sub(args[:old_domain], args[:new_domain]))
```

Comment:
This seems unlikely to be necessary, but is it worth protecting against accidental modification of the user side of the email address in weird cases where it overlaps with the domain?  Like, what if you have `penny.eduperson@ny.edu`?

Maybe use `sub("@#{args[:old_domain]}", "@#{args[:new_domain]}")`?

Also, do we need to worry about casing conflict in the `sub`?

Reply:
That's a great point on the '@'.  I will update.

I'm not sure I understand the question on the casing conflict.

Reply:
Is it possible for `sub` to fail to replace `Thomas.Robertson@Quill.org` if `old_domain` is `quill.org` because there's a case mismatch between the on-record email address and the specified arguments for the rake task?

Reply:
That's another good point.  While, I think all persisted user emails are downcased, the rake task wouldn't work if `new_domain` were capitalized in anyway.  I'll just downcase both for clarity.